packages/wrangler/templates/middleware/middleware-ensure-req-body-drained.ts

```diff
@@ -0,0 +1,14 @@
+import type { Middleware } from "./common";
+
+const drainBody: Middleware = async (request, env, _ctx, middlewareCtx) => {
+	try {
+		return await middlewareCtx.next(request, env);
+	} finally {
+		if (request.body !== null && !request.bodyUsed) {
+			const reader = request.body.getReader();
+			while (!(await reader.read()).done) {}
```

Comment:
Do we need try..catch around this await?

Reply:
That sounds like a good idea.
Out of curiosity, in what way might this fail? Like if the client disconnects during the drain?

Reply:
I actually don't know. But an async loop reading chunks of a stream seems like something that _could_ fail. My original ctx.waitUntil handled this by allowing the arrayBuffer promise to reject without affecting the response promise. We should handle it to be safe.